src/main/java/org/broadinstitute/hellbender/tools/dataflow/pipelines/ReadsPreprocessingPipeline.java

```diff
@@ -83,14 +84,24 @@ protected void setupPipeline( Pipeline pipeline ) {
                     : IntervalUtils.getAllIntervalsForReference(readsHeader.getSequenceDictionary());
 
             final PCollectionView<SAMFileHeader> headerSingleton = ReadsDataflowSource.getHeaderView(pipeline, readsHeader);
-            final PCollection<GATKRead> initialReads = readsDataflowSource.getReadPCollection(intervals);
 
             final OpticalDuplicateFinder finder = opticalDuplicatesArgumentCollection.READ_NAME_REGEX != null ?
                     new OpticalDuplicateFinder(opticalDuplicatesArgumentCollection.READ_NAME_REGEX, opticalDuplicatesArgumentCollection.OPTICAL_DUPLICATE_PIXEL_DISTANCE, null) : null;
             final PCollectionView<OpticalDuplicateFinder> finderPcolView = pipeline.apply(Create.of(finder)).apply(View.<OpticalDuplicateFinder>asSingleton());
 
+            // we read from shards that span this many bp
+            int basesPerShard = 1_000_000;
+            // and then output shards that span this many
+            int outputBasesPerWorkUnit = 5_000;
+
             // Apply MarkDuplicates to produce updated GATKReads.
-            final PCollection<GATKRead> markedReads = initialReads.apply(new MarkDuplicates(headerSingleton, finderPcolView));
```

Comment:
There should be an option to use the original MarkDuplicates, since Spark/Hadoop only works with this version. MarkDuplicatesFromShardsDataflowTransform is probably not needed for running on Hadoop since Hadoop will already allocate shards of the input BAMs to workers.